### PR TITLE
py-neurodamus: bump to 3.0a1 for a pre-release

### PR DESCRIPTION
**Pull request by WeinaJi**

### CODE REVIEW

File: bluebrain/repo-bluebrain/packages/py-neurodamus/package.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ class PyNeurodamus(PythonPackage):
     git = "https://github.com/BlueBrain/neurodamus.git"
 
     version("develop", branch="main")
+    version("2.17.1", commit="493e8ea")
```
Comment (pramodk) [inline bluebrain/repo-bluebrain/packages/py-neurodamus/package.py]: `nitpick`: if this is based on a commit, it should be added as `alpha` or something similar?

Comment (WeinaJi) [reply bluebrain/repo-bluebrain/packages/py-neurodamus/package.py]: I am wondering. I see neuron uses `a[number]` but py-libsonata uses normal versions. What's the rule?

Comment (WeinaJi) [reply bluebrain/repo-bluebrain/packages/py-neurodamus/package.py]: or `2.17.0.1`?

Comment (matz-e) [reply bluebrain/repo-bluebrain/packages/py-neurodamus/package.py]: Spack handles alpha versions very poorly. Don't do that :)

@WeinaJi `2.17.0.1` would work, or adding the date, like `2.17.0-20240122`

File: bluebrain/repo-bluebrain/packages/py-neurodamus/package.py [hunk 2]
```diff
@@ -12,6 +12,7 @@ class PyNeurodamus(PythonPackage):
     git = "https://github.com/BlueBrain/neurodamus.git"
 
     version("develop", branch="main")
+    version("3.0a1", tag="3.0.0-alpha")
```
Comment (pramodk) [inline bluebrain/repo-bluebrain/packages/py-neurodamus/package.py:15]: @WeinaJi : just be aware of how you tag / name the version

```bash
$ spack-python

>>> from spack.version import Version as V;
>>> print(V('3.0') > V('3.0a1'))
False
>>> print(V('3.0.0') > V('3.0a1'))
True
>>> print(V('3.0.0') > V('3.0.a1'))
True
```

i.e. if you create `3.0` as next version in spack then it won't be greater than `3.0a`!

Comment (pramodk) [reply bluebrain/repo-bluebrain/packages/py-neurodamus/package.py:15]: here is our old Slack discussion: https://bluebrainproject.slack.com/archives/C023LSDKJUU/p1680617713058199?thread_ts=1680610388.183389&cid=C023LSDKJUU

Comment (WeinaJi) [reply bluebrain/repo-bluebrain/packages/py-neurodamus/package.py:15]: The next stable version will be 3.0.0. Why `3.0` ? We always use the versioning `x.y.z`.

Comment (pramodk) [reply bluebrain/repo-bluebrain/packages/py-neurodamus/package.py:15]: yes...just want to highlight this naming aspect!

**Review by pramodk — APPROVED** [review]

**Review by jorblancoa — APPROVED** [review]

**Review by matz-e — APPROVED** [review]

**Pull request merged**